slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbConnection.java

```diff
@@ -61,12 +77,40 @@ public static void configureJdbi(final Jdbi jdbi) {
   }
 
   private static HikariDataSource createDataSource(
-      final String jdbcUrl, final String username, final String password) {
-    final HikariDataSource dataSource = new HikariDataSource();
-    dataSource.setJdbcUrl(jdbcUrl);
-    dataSource.setUsername(username);
-    dataSource.setPassword(password);
-    dataSource.addDataSourceProperty(PG_SOCKET_TIMEOUT_PARAM, DEFAULT_PG_SOCKET_TIMEOUT_SECONDS);
-    return dataSource;
+      final String jdbcUrl,
+      final String username,
+      final String password,
+      final Path hikariConfigurationFile) {
+    final Properties hikariConfigurationProperties =
+        loadHikariConfigurationProperties(hikariConfigurationFile);
+
+    final HikariConfig hikariConfig = new HikariConfig(hikariConfigurationProperties);
+    hikariConfig.setJdbcUrl(jdbcUrl);
+    if (!isEmpty(username)) {
+      hikariConfig.setUsername(username);
+    }
+    if (!isEmpty(password)) {
+      hikariConfig.setPassword(password);
+    }
+
+    return new HikariDataSource(hikariConfig);
+  }
+
+  @VisibleForTesting
+  static Properties loadHikariConfigurationProperties(final Path hikariConfigurationFile) {
+    final Properties hikariConfigurationProperties = new Properties();
+    if (hikariConfigurationFile != null) {
+      try (FileInputStream inputStream = new FileInputStream(hikariConfigurationFile.toFile())) {
+        hikariConfigurationProperties.load(inputStream);
+      } catch (final IOException e) {
+        throw new UncheckedIOException(e);
```

Comment:
Will the application still start if we get an error?

Reply:
No, application fails to start. when exception is thrown.
~~~
./web3signer --key-store-path=./large_keys -l debug --swagger-ui-enabled=true eth2 --slashing-protection-db-url="jdbc:postgresql://localhost/web3signer" --slashing-protection-db-username=postgres --slashing-protection-db-password=postgres --slashing-protection-pruning-enabled=true --slashing-protection-db-pool-configuration-file ./hikari.propertiesss
2021-06-08 09:46:16.332+10:00 | main | INFO  | Web3SignerApp | Web3Signer has started with args --key-store-path=./large_keys,-l,debug,--swagger-ui-enabled=true,eth2,--slashing-protection-db-url=jdbc:postgresql://localhost/web3signer,--slashing-protection-db-username=postgres,--slashing-protection-db-password=postgres,--slashing-protection-pruning-enabled=true,--slashing-protection-db-pool-configuration-file,./hikari.propertiesss
2021-06-08 09:46:16.337+10:00 | main | INFO  | Web3SignerApp | Version = web3signer/v21.3.0+7-g78a01b9/osx-x86_64/adoptopenjdk-java-16
Failed to initialize Web3Signer
Cause: java.io.FileNotFoundException: ./hikari.propertiesss (No such file or directory)
java.io.UncheckedIOException: java.io.FileNotFoundException: ./hikari.propertiesss (No such file or directory)
	at tech.pegasys.web3signer.slashingprotection.DbConnection.loadHikariConfigurationProperties(DbConnection.java:100)
	at tech.pegasys.web3signer.slashingprotection.DbConnection.createDataSource(DbConnection.java:83)
	at tech.pegasys.web3signer.slashingprotection.DbConnection.createConnection(DbConnection.java:47)
	at tech.pegasys.web3signer.slashingprotection.SlashingProtectionFactory.createSlashingProtection(SlashingProtectionFactory.java:31)
	at tech.pegasys.web3signer.core.Eth2Runner.createSlashingProtection(Eth2Runner.java:93)
	at tech.pegasys.web3signer.core.Eth2Runner.<init>(Eth2Runner.java:82)
	at tech.pegasys.web3signer.commandline.subcommands.Eth2SubCommand.createRunner(Eth2SubCommand.java:50)
	at tech.pegasys.web3signer.commandline.subcommands.ModeSubCommand.run(ModeSubCommand.java:28)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1919)
	at picocli.CommandLine.access$1100(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2332)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2326)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2291)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2159)
	at picocli.CommandLine.execute(CommandLine.java:2058)
	at tech.pegasys.web3signer.commandline.CommandlineParser.parseCommandLine(CommandlineParser.java:85)
	at tech.pegasys.web3signer.Web3SignerApp.executeWithEnvironment(Web3SignerApp.java:50)
	at tech.pegasys.web3signer.Web3SignerApp.main(Web3SignerApp.java:35)
Caused by: java.io.FileNotFoundException: ./hikari.propertiesss (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:211)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:153)
	at tech.pegasys.web3signer.slashingprotection.DbConnection.loadHikariConfigurationProperties(DbConnection.java:97)
	... 17 more
~~~